packages/stellar-axelar-std-derive/src/into_event.rs

```diff
@@ -55,9 +55,12 @@ pub fn into_event(input: &DeriveInput) -> proc_macro2::TokenStream {
             // Parse data from Val to the corresponding types,
             // and assign them to a variable with the same name as the struct field
             // E.g. let message = Message::try_from_val(env, &data.get(0));
-            // `data` is required to be a `Vec<Val>`
             let data = Vec::<Val>::try_from_val(env, &data)
-                .expect("invalid data format");
+                .unwrap_or_else(|_| {
+                    let mut vec = Vec::<Val>::new(env);
+                    vec.push_back(data);
+                    vec
+                });
```

Comment:
This implicit fallback is bad practice. Consider how it can cause unintentional issues. E.g. if another event accidentally used the value directly instead of a vector, then this fallback could cause it to be decoded correctly, even though the data field wasn't formatted as expected.